lib/active_remote/typecasting/date_typecaster.rb

```diff
@@ -0,0 +1,10 @@
+module ActiveRemote
+  module Typecasting
+    class DateTypecaster
+      def self.call(value)
+        value.to_date if value.respond_to? :to_date
+      rescue NoMethodError, ArgumentError
```

Comment:
Actually, it's considerably slower:

```
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]
                      user     system      total        real
rescue:          24.840000   0.480000  25.320000 ( 25.459208)
condition:        1.400000   0.000000   1.400000 (  1.418840)
rescue/condition 17.742857        Inf        NaN ( 17.943681)
```

Based on http://greyblake.com/blog/2012/09/02/ruby-perfomance-tricks/